### PR TITLE
ROX-12623: ListCollections rest endpoint

### PR DESCRIPTION
**Pull request by keyallis**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: central/resourcecollection/service/service_impl.go [hunk 1]
```diff
@@ -15,15 +15,21 @@ import (
 	"github.com/stackrox/rox/pkg/grpc/authz/or"
 	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
 	"github.com/stackrox/rox/pkg/grpc/authz/user"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/search/paginated"
 	"google.golang.org/grpc"
 )
 
+const (
+	maxCollectionsReturned = 1000
```
Comment (md2119) [inline central/resourcecollection/service/service_impl.go:24]: nit: `defaultPageSize`

File: central/resourcecollection/service/service_impl.go [hunk 2]
```diff
@@ -81,3 +87,27 @@ func (s *serviceImpl) getCollection(ctx context.Context, id string) (*v1.GetColl
 		Deployments: nil,
 	}, nil
 }
+
+func (s *serviceImpl) ListCollections(ctx context.Context, request *v1.ListCollectionsRequest) (*v1.ListCollectionsResponse, error) {
+	if !features.ObjectCollections.Enabled() {
+		return nil, nil
```
Comment (md2119) [inline central/resourcecollection/service/service_impl.go:93]: return error

**Review by md2119 — APPROVED** [review]

**Review by charmik-redhat — APPROVED** [review]
lgtm, good to merge after addressing the one comment

**Pull request merged**